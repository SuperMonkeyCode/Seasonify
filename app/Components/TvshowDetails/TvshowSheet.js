/* @flow */
import { Content, Form, Item, Input, Icon, List, ListItem, Text } from 'native-base';
import { pure } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  name: string,
  year: number,
  localizedName?: string,
  isEditing: boolean,
  edit: Object,
  onChangeName: Function,
  /* connect */
  /* HOC */
};

const enhance = pure;

function TvshowSheet({
  name,
  year,
  localizedName,
  isEditing,
  edit,
  onChangeName,
}: Props) {
  return (
    <Content>
      <Form>
        <Item regular>
          {isEditing && <Icon name="create" />}
          <Input
            disabled={!isEditing}
            placeHolder="Name"
            value={'name' in edit ? edit.name : name}
            onChangeText={onChangeName}
            autoCapitalize="words"
          />
        </Item>
      </Form>
      <List>
        {localizedName &&
          <ListItem>
            <Text note>
              {localizedName}
            </Text>
          </ListItem>}
        <ListItem>
          <Text note>
            {year}
          </Text>
        </ListItem>
      </List>
    </Content>
  );
}

export default enhance(TvshowSheet);
