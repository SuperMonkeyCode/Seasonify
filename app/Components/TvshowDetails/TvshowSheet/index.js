/* @flow */
export { default } from './TvshowSheet';
