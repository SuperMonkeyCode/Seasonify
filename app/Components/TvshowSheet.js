import React from 'react';
import {
  Content,
  Form,
  Item,
  Input,
  Icon,
  List,
  ListItem,
  Text,
} from 'native-base';

export default function TvshowSheet({
  name,
  year,
  localizedName,
  isEditing,
  edit,
  onChangeName,
}) {
  return (
    <Content>
      <Form>
        <Item regular>
          {isEditing && <Icon name="create" />}
          <Input
            disabled={!isEditing}
            placeHolder="Name"
            value={'name' in edit ? edit.name : name}
            onChangeText={onChangeName}
            autoCapitalize="words"
          />
        </Item>
      </Form>
      <List>
        {localizedName &&
          <ListItem>
            <Text note>{localizedName}</Text>
          </ListItem>}
        <ListItem>
          <Text note>{year}</Text>
        </ListItem>
      </List>
    </Content>
  );
}
