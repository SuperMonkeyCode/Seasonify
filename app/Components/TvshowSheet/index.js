/* @flow */
import TvshowSheet from './TvshowSheet';

export default TvshowSheet;
