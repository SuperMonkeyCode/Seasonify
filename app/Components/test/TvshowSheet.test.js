import React from 'react';
import TvshowSheet from '../TvshowSheet';

function setup(specificProps = {}) {
  const props = {
    name: "Turn : Washington's Spies",
    localizedName: undefined,
    year: 2014,
    isEditing: false,
    edit: {},
    onChangeName: jest.fn(),
    ...specificProps,
  };
  const component = shallow(<TvshowSheet {...props} />);
  return {
    component,
    props,
  };
}

describe('Rendering', () => {
  it('should match', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });

  it('should match, when in editing mode', () => {
    const { component } = setup({ isEditing: true });
    expect(component).toMatchSnapshot();
  });

  it('should match, when form is edited', () => {
    const { component } = setup({
      isEditing: true,
      edit: { name: 'TURN', year: '2015' },
    });
    expect(component).toMatchSnapshot();
  });

  it('should match, when localized name is present', () => {
    const { component } = setup({ localizedName: 'Turn' });
    expect(component).toMatchSnapshot();
  });
});

describe('Events', () => {
  it('should call onChangeName, when changing text', () => {
    const { component, props } = setup();
    component.find('Styled(Input)').simulate('changeText');
    expect(props.onChangeName).toBeCalled();
  });
});
