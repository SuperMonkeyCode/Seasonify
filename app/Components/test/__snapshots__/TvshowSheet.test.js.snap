// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should match 1`] = `
<Styled(Content)>
  <Styled(Form)>
    <Styled(Item)
      regular={true}
    >
      <Styled(Input)
        autoCapitalize="words"
        disabled={true}
        onChangeText={[Function]}
        placeHolder="Name"
        value="Turn : Washington's Spies"
      />
    </Styled(Item)>
  </Styled(Form)>
  <Styled(List)>
    <Styled(ListItem)>
      <Styled(Text)
        note={true}
      >
        2014
      </Styled(Text)>
    </Styled(ListItem)>
  </Styled(List)>
</Styled(Content)>
`;

exports[`Rendering should match, when form is edited 1`] = `
<Styled(Content)>
  <Styled(Form)>
    <Styled(Item)
      regular={true}
    >
      <Styled(Icon)
        name="create"
      />
      <Styled(Input)
        autoCapitalize="words"
        disabled={false}
        onChangeText={[Function]}
        placeHolder="Name"
        value="TURN"
      />
    </Styled(Item)>
  </Styled(Form)>
  <Styled(List)>
    <Styled(ListItem)>
      <Styled(Text)
        note={true}
      >
        2014
      </Styled(Text)>
    </Styled(ListItem)>
  </Styled(List)>
</Styled(Content)>
`;

exports[`Rendering should match, when in editing mode 1`] = `
<Styled(Content)>
  <Styled(Form)>
    <Styled(Item)
      regular={true}
    >
      <Styled(Icon)
        name="create"
      />
      <Styled(Input)
        autoCapitalize="words"
        disabled={false}
        onChangeText={[Function]}
        placeHolder="Name"
        value="Turn : Washington's Spies"
      />
    </Styled(Item)>
  </Styled(Form)>
  <Styled(List)>
    <Styled(ListItem)>
      <Styled(Text)
        note={true}
      >
        2014
      </Styled(Text)>
    </Styled(ListItem)>
  </Styled(List)>
</Styled(Content)>
`;

exports[`Rendering should match, when localized name is present 1`] = `
<Styled(Content)>
  <Styled(Form)>
    <Styled(Item)
      regular={true}
    >
      <Styled(Input)
        autoCapitalize="words"
        disabled={true}
        onChangeText={[Function]}
        placeHolder="Name"
        value="Turn : Washington's Spies"
      />
    </Styled(Item)>
  </Styled(Form)>
  <Styled(List)>
    <Styled(ListItem)>
      <Styled(Text)
        note={true}
      >
        Turn
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)>
      <Styled(Text)
        note={true}
      >
        2014
      </Styled(Text)>
    </Styled(ListItem)>
  </Styled(List)>
</Styled(Content)>
`;
